In [1]:
import pandas as pd

In [41]:
import pandas as pd

In [42]:
from pandas import read_csv

№2.1.

1. Скачать данные по ссылке https://www.kaggle.com/datasets/ionaskel/laptop-prices. Если ссылка не открывается - найти на kaggle этот файл у другого пользователя.

Или файл прикреплен к семинару. Можно открыть по прямой ссылке из notebook

2. Считать данные с помощью pandas

3. Вывести на экран первые 5 строк

4. Посмотреть на описание признаков и на их содержание: каого типа поля и данные.

In [63]:
df = pd.read_csv(r"C:\Users\HONOR\OneDrive\Рабочий стол\laptop_price.csv", encoding = "cp1251")
df

laptop_ID Company                              Product  \
0             1   Apple                          MacBook Pro   
1             2   Apple                          Macbook Air   
2             3      HP                               250 G6   
3             4   Apple                          MacBook Pro   
4             5   Apple                          MacBook Pro   
...         ...     ...                                  ...   
1298       1316  Lenovo                       Yoga 500-14ISK   
1299       1317  Lenovo                       Yoga 900-13ISK   
1300       1318  Lenovo                   IdeaPad 100S-14IBR   
1301       1319      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon   
1302       1320    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)   

                TypeName  Inches                            ScreenResolution  \
0              Ultrabook    13.3          IPS Panel Retina Display 2560x1600   
1              Ultrabook    13.3                                    1440x900   
2               Notebook    15.6                           Full HD 1920x1080   
3              Ultrabook    15.4          IPS Panel Retina Display 2880x1800   
4              Ultrabook    13.3          IPS Panel Retina Display 2560x1600   
...                  ...     ...                                         ...   
1298  2 in 1 Convertible    14.0   IPS Panel Full HD / Touchscreen 1920x1080   
1299  2 in 1 Convertible    13.3  IPS Panel Quad HD+ / Touchscreen 3200x1800   
1300            Notebook    14.0                                    1366x768   
1301            Notebook    15.6                                    1366x768   
1302            Notebook    15.6                                    1366x768   

                                       Cpu   Ram               Memory  \
0                     Intel Core i5 2.3GHz   8GB            128GB SSD   
1                     Intel Core i5 1.8GHz   8GB  128GB Flash Storage   
2               Intel Core i5 7200U 2.5GHz   8GB            256GB SSD   
3                     Intel Core i7 2.7GHz  16GB            512GB SSD   
4                     Intel Core i5 3.1GHz   8GB            256GB SSD   
...                                    ...   ...                  ...   
1298            Intel Core i7 6500U 2.5GHz   4GB            128GB SSD   
1299            Intel Core i7 6500U 2.5GHz  16GB            512GB SSD   
1300  Intel Celeron Dual Core N3050 1.6GHz   2GB   64GB Flash Storage   
1301            Intel Core i7 6500U 2.5GHz   6GB              1TB HDD   
1302  Intel Celeron Dual Core N3050 1.6GHz   4GB            500GB HDD   

                               Gpu       OpSys  Weight  Price_euros  
0     Intel Iris Plus Graphics 640       macOS  1.37kg      1339.69  
1           Intel HD Graphics 6000       macOS  1.34kg       898.94  
2            Intel HD Graphics 620       No OS  1.86kg       575.00  
3               AMD Radeon Pro 455       macOS  1.83kg      2537.45  
4     Intel Iris Plus Graphics 650       macOS  1.37kg      1803.60  
...                            ...         ...     ...          ...  
1298         Intel HD Graphics 520  Windows 10   1.8kg       638.00  
1299         Intel HD Graphics 520  Windows 10   1.3kg      1499.00  
1300             Intel HD Graphics  Windows 10   1.5kg       229.00  
1301            AMD Radeon R5 M330  Windows 10  2.19kg       764.00  
1302             Intel HD Graphics  Windows 10   2.2kg       369.00  

[1303 rows x 13 columns]

In [64]:
df.head()

laptop_ID Company      Product   TypeName  Inches  \
0          1   Apple  MacBook Pro  Ultrabook    13.3   
1          2   Apple  Macbook Air  Ultrabook    13.3   
2          3      HP       250 G6   Notebook    15.6   
3          4   Apple  MacBook Pro  Ultrabook    15.4   
4          5   Apple  MacBook Pro  Ultrabook    13.3   

                     ScreenResolution                         Cpu   Ram  \
0  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                            1440x900        Intel Core i5 1.8GHz   8GB   
2                   Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3  IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4  IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   

                Memory                           Gpu  OpSys  Weight  \
0            128GB SSD  Intel Iris Plus Graphics 640  macOS  1.37kg   
1  128GB Flash Storage        Intel HD Graphics 6000  macOS  1.34kg   
2            256GB SSD         Intel HD Graphics 620  No OS  1.86kg   
3            512GB SSD            AMD Radeon Pro 455  macOS  1.83kg   
4            256GB SSD  Intel Iris Plus Graphics 650  macOS  1.37kg   

   Price_euros  
0      1339.69  
1       898.94  
2       575.00  
3      2537.45  
4      1803.60

In [66]:
df.tail()

laptop_ID Company                              Product  \
1298       1316  Lenovo                       Yoga 500-14ISK   
1299       1317  Lenovo                       Yoga 900-13ISK   
1300       1318  Lenovo                   IdeaPad 100S-14IBR   
1301       1319      HP  15-AC110nv (i7-6500U/6GB/1TB/Radeon   
1302       1320    Asus  X553SA-XX031T (N3050/4GB/500GB/W10)   

                TypeName  Inches                            ScreenResolution  \
1298  2 in 1 Convertible    14.0   IPS Panel Full HD / Touchscreen 1920x1080   
1299  2 in 1 Convertible    13.3  IPS Panel Quad HD+ / Touchscreen 3200x1800   
1300            Notebook    14.0                                    1366x768   
1301            Notebook    15.6                                    1366x768   
1302            Notebook    15.6                                    1366x768   

                                       Cpu   Ram              Memory  \
1298            Intel Core i7 6500U 2.5GHz   4GB           128GB SSD   
1299            Intel Core i7 6500U 2.5GHz  16GB           512GB SSD   
1300  Intel Celeron Dual Core N3050 1.6GHz   2GB  64GB Flash Storage   
1301            Intel Core i7 6500U 2.5GHz   6GB             1TB HDD   
1302  Intel Celeron Dual Core N3050 1.6GHz   4GB           500GB HDD   

                        Gpu       OpSys  Weight  Price_euros  
1298  Intel HD Graphics 520  Windows 10   1.8kg        638.0  
1299  Intel HD Graphics 520  Windows 10   1.3kg       1499.0  
1300      Intel HD Graphics  Windows 10   1.5kg        229.0  
1301     AMD Radeon R5 M330  Windows 10  2.19kg        764.0  
1302      Intel HD Graphics  Windows 10   2.2kg        369.0

In [67]:
df.head(17)

laptop_ID Company          Product   TypeName  Inches  \
0           1   Apple      MacBook Pro  Ultrabook    13.3   
1           2   Apple      Macbook Air  Ultrabook    13.3   
2           3      HP           250 G6   Notebook    15.6   
3           4   Apple      MacBook Pro  Ultrabook    15.4   
4           5   Apple      MacBook Pro  Ultrabook    13.3   
5           6    Acer         Aspire 3   Notebook    15.6   
6           7   Apple      MacBook Pro  Ultrabook    15.4   
7           8   Apple      Macbook Air  Ultrabook    13.3   
8           9    Asus  ZenBook UX430UN  Ultrabook    14.0   
9          10    Acer          Swift 3  Ultrabook    14.0   
10         11      HP           250 G6   Notebook    15.6   
11         12      HP           250 G6   Notebook    15.6   
12         13   Apple      MacBook Pro  Ultrabook    15.4   
13         14    Dell    Inspiron 3567   Notebook    15.6   
14         15   Apple      MacBook 12"  Ultrabook    12.0   
15         16   Apple      MacBook Pro  Ultrabook    13.3   
16         17    Dell    Inspiron 3567   Notebook    15.6   

                      ScreenResolution                         Cpu   Ram  \
0   IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
1                             1440x900        Intel Core i5 1.8GHz   8GB   
2                    Full HD 1920x1080  Intel Core i5 7200U 2.5GHz   8GB   
3   IPS Panel Retina Display 2880x1800        Intel Core i7 2.7GHz  16GB   
4   IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz   8GB   
5                             1366x768     AMD A9-Series 9420 3GHz   4GB   
6   IPS Panel Retina Display 2880x1800        Intel Core i7 2.2GHz  16GB   
7                             1440x900        Intel Core i5 1.8GHz   8GB   
8                    Full HD 1920x1080  Intel Core i7 8550U 1.8GHz  16GB   
9          IPS Panel Full HD 1920x1080  Intel Core i5 8250U 1.6GHz   8GB   
10                            1366x768  Intel Core i5 7200U 2.5GHz   4GB   
11                   Full HD 1920x1080    Intel Core i3 6006U 2GHz   4GB   
12  IPS Panel Retina Display 2880x1800        Intel Core i7 2.8GHz  16GB   
13                   Full HD 1920x1080    Intel Core i3 6006U 2GHz   4GB   
14  IPS Panel Retina Display 2304x1440      Intel Core M m3 1.2GHz   8GB   
15  IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz   8GB   
16                   Full HD 1920x1080  Intel Core i7 7500U 2.7GHz   8GB   

                 Memory                           Gpu       OpSys  Weight  \
0             128GB SSD  Intel Iris Plus Graphics 640       macOS  1.37kg   
1   128GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
2             256GB SSD         Intel HD Graphics 620       No OS  1.86kg   
3             512GB SSD            AMD Radeon Pro 455       macOS  1.83kg   
4             256GB SSD  Intel Iris Plus Graphics 650       macOS  1.37kg   
5             500GB HDD                 AMD Radeon R5  Windows 10   2.1kg   
6   256GB Flash Storage       Intel Iris Pro Graphics    Mac OS X  2.04kg   
7   256GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
8             512GB SSD          Nvidia GeForce MX150  Windows 10   1.3kg   
9             256GB SSD        Intel UHD Graphics 620  Windows 10   1.6kg   
10            500GB HDD         Intel HD Graphics 620       No OS  1.86kg   
11            500GB HDD         Intel HD Graphics 520       No OS  1.86kg   
12            256GB SSD            AMD Radeon Pro 555       macOS  1.83kg   
13            256GB SSD            AMD Radeon R5 M430  Windows 10   2.2kg   
14            256GB SSD         Intel HD Graphics 615       macOS  0.92kg   
15            256GB SSD  Intel Iris Plus Graphics 640       macOS  1.37kg   
16            256GB SSD            AMD Radeon R5 M430  Windows 10   2.2kg   

    Price_euros  
0       1339.69  
1        898.94  
2        575.00  
3       2537.45  
4       1803.60  
5        400.00  
6       2139.97  
7    

№2.2.

Проведите первичный анализ данных:

1. Изучите типы данных

2. Найдите количество пропущенных ячеек в данных (non-null)

3. Посчитайте основные статистики по всем признакам и поизучайте их

In [68]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1303 entries, 0 to 1302
Data columns (total 13 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   laptop_ID         1303 non-null   int64  
 1   Company           1303 non-null   object 
 2   Product           1303 non-null   object 
 3   TypeName          1303 non-null   object 
 4   Inches            1303 non-null   float64
 5   ScreenResolution  1303 non-null   object 
 6   Cpu               1303 non-null   object 
 7   Ram               1303 non-null   object 
 8   Memory            1303 non-null   object 
 9   Gpu               1303 non-null   object 
 10  OpSys             1303 non-null   object 
 11  Weight            1303 non-null   object 
 12  Price_euros       1303 non-null   float64
dtypes: float64(2), int64(1), object(10)
memory usage: 132.5+ KB


In [73]:
df.describe()

laptop_ID       Inches  Price_euros
count  1303.000000  1303.000000  1303.000000
mean    660.155794    15.017191  1123.686992
std     381.172104     1.426304   699.009043
min       1.000000    10.100000   174.000000
25%     331.500000    14.000000   599.000000
50%     659.000000    15.600000   977.000000
75%     990.500000    15.600000  1487.880000
max    1320.000000    18.400000  6099.000000

In [74]:
df.describe(include=['object'])

Company Product  TypeName   ScreenResolution  \
count     1303    1303      1303               1303   
unique      19     618         6                 40   
top       Dell  XPS 13  Notebook  Full HD 1920x1080   
freq       297      30       727                507   

                               Cpu   Ram     Memory                    Gpu  \
count                         1303  1303       1303                   1303   
unique                         118     9         39                    110   
top     Intel Core i5 7200U 2.5GHz   8GB  256GB SSD  Intel HD Graphics 620   
freq                           190   619        412                    281   

             OpSys Weight  
count         1303   1303  
unique           9    179  
top     Windows 10  2.2kg  
freq          1072    121

№2.3

1. Ноутбуков от какой компании больше всего в наборе данных?

2. Какая минимальная и максимальная стоимость у ноутбуков в данных?

Усложнение: Вопрос касается только ноутбуков.

3. Какой самый дорогой ноутбук в данных?

Сделать с отдельной переменной для булевой маске по цене или логическое условие в одну строку (без дополнительной переменной)
Выведите все характеристики только по этому ноутбуку
Если таких ноутбуков несколько, то выводите их всех

In [78]:
df['Company'].value_counts()

Company
Dell         297
Lenovo       297
HP           274
Asus         158
Acer         103
MSI           54
Toshiba       48
Apple         21
Samsung        9
Razer          7
Mediacom       7
Microsoft      6
Xiaomi         4
Vero           4
Chuwi          3
Google         3
Fujitsu        3
LG             3
Huawei         2
Name: count, dtype: int64

In [79]:
type(df['Company'])

pandas.core.series.Series

In [89]:
df['Company'].value_counts(normalize=True)


Company
Dell         0.227936
Lenovo       0.227936
HP           0.210284
Asus         0.121259
Acer         0.079048
MSI          0.041443
Toshiba      0.036838
Apple        0.016117
Samsung      0.006907
Razer        0.005372
Mediacom     0.005372
Microsoft    0.004605
Xiaomi       0.003070
Vero         0.003070
Chuwi        0.002302
Google       0.002302
Fujitsu      0.002302
LG           0.002302
Huawei       0.001535
Name: proportion, dtype: float64

In [103]:
df[df['TypeName'] == 'Notebook'].describe()

laptop_ID      Inches  Price_euros
count   727.000000  727.000000   727.000000
mean    643.276479   15.330536   782.078624
std     381.841542    0.991550   443.972330
min       3.000000   11.600000   196.000000
25%     312.500000   15.600000   469.005000
50%     617.000000   15.600000   691.000000
75%     982.000000   15.600000   979.000000
max    1320.000000   17.300000  4899.000000

In [108]:
df[df['TypeName'] == 'Notebook']['Price_euros'].max()

4899.0

In [109]:
df[df['TypeName'] == 'Notebook']['Price_euros'].min()

196.0

In [112]:
df[df['Price_euros'] == df[df['TypeName'] == 'Notebook']['Price_euros'].max()]

laptop_ID Company       Product  TypeName  Inches  \
610        617  Lenovo  Thinkpad P51  Notebook    15.6   

                    ScreenResolution                            Cpu   Ram  \
610  IPS Panel 4K Ultra HD 3840x2160  Intel Xeon E3-1535M v6 3.1GHz  32GB   

      Memory                   Gpu       OpSys Weight  Price_euros  
610  1TB SSD  Nvidia Quadro M2200M  Windows 10  2.5kg       4899.0

In [115]:
priceMax = df[df['TypeName'] == 'Notebook']['Price_euros'].max()
priceMax

4899.0

In [116]:
df[df['Price_euros'] == priceMax]

laptop_ID Company       Product  TypeName  Inches  \
610        617  Lenovo  Thinkpad P51  Notebook    15.6   

                    ScreenResolution                            Cpu   Ram  \
610  IPS Panel 4K Ultra HD 3840x2160  Intel Xeon E3-1535M v6 3.1GHz  32GB   

      Memory                   Gpu       OpSys Weight  Price_euros  
610  1TB SSD  Nvidia Quadro M2200M  Windows 10  2.5kg       4899.0

№2.4
1. Найдите ноутбуки с самой маленькой диагональю в данных?
- Выведите только следующие характеристики: Компания, Типа ноутбука, Диагональ, Стоимость
- Если таких ноутбуков несколько, то выводите их всех
2. Сколько стоит самый дорогой ноутбук у компании HP?
3. Как много ноутбуков Ultrabook с 8GB RAM?
- Найдите сколько таких ультрабуков с 8GB ОЗУ в процентном соотношении относительно всех ультрабуков

In [117]:
df['Inches'].describe()

count    1303.000000
mean       15.017191
std         1.426304
min        10.100000
25%        14.000000
50%        15.600000
75%        15.600000
max        18.400000
Name: Inches, dtype: float64

In [118]:
df['Inches'].min()

10.1

вывели все ноутбуки с мин.диагональю:

In [122]:
df[df['Inches'] == df['Inches'].min()]

laptop_ID Company    Product            TypeName  Inches  \
50           51  Lenovo  Yoga Book  2 in 1 Convertible    10.1   
1082       1097  Lenovo  Yoga Book  2 in 1 Convertible    10.1   
1114       1129  Lenovo  Yoga Book  2 in 1 Convertible    10.1   
1126       1141  Lenovo  Yoga Book  2 in 1 Convertible    10.1   

                     ScreenResolution                          Cpu  Ram  \
50    IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   
1082  IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   
1114  IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   
1126  IPS Panel Touchscreen 1920x1200  Intel Atom x5-Z8550 1.44GHz  4GB   

                  Memory                    Gpu       OpSys  Weight  \
50    64GB Flash Storage  Intel HD Graphics 400     Android  0.69kg   
1082  64GB Flash Storage  Intel HD Graphics 400  Windows 10  0.69kg   
1114  64GB Flash Storage  Intel HD Graphics 400     Android  0.69kg   
1126  64GB Flash Storage  Intel HD Graphics 400  Windows 10  0.69kg   

      Price_euros  
50         319.00  
1082       646.27  
1114       549.00  
1126       479.00

In [123]:
df['Price_euros'].max()

6099.0

In [124]:
priceMaxHP = df[df['Company'] == 'HP']['Price_euros'].max()
priceMaxHP

4389.0

In [128]:
df[df['TypeName'] == 'Ultrabook'].value_counts()

laptop_ID  Company  Product                 TypeName   Inches  ScreenResolution                    Cpu                          Ram  Memory               Gpu                           OpSys       Weight  Price_euros
1          Apple    MacBook Pro             Ultrabook  13.3    IPS Panel Retina Display 2560x1600  Intel Core i5 2.3GHz         8GB  128GB SSD            Intel Iris Plus Graphics 640  macOS       1.37kg  1339.69        1
833        Lenovo   ThinkPad X1             Ultrabook  14.0    IPS Panel Full HD 1920x1080         Intel Core i5 7200U 2.5GHz   8GB  256GB Flash Storage  Intel HD Graphics 620         Windows 10  1.13kg  1875.00        1
854        HP       EliteBook 840           Ultrabook  14.0    Full HD 1920x1080                   Intel Core i7 6500U 2.5GHz   8GB  512GB SSD            Intel HD Graphics 520         Windows 7   1.54kg  1870.00        1
857        Asus     ZenBook UX310UQ-GL026T  Ultrabook  13.3    IPS Panel Full HD 1920x1080         Intel Core i5 6200U 2.

In [149]:
df[(df['TypeName'] == 'Ultrabook') & (df['Ram'] == '8GB')]

laptop_ID Company               Product   TypeName  Inches  \
0             1   Apple           MacBook Pro  Ultrabook    13.3   
1             2   Apple           Macbook Air  Ultrabook    13.3   
4             5   Apple           MacBook Pro  Ultrabook    13.3   
7             8   Apple           Macbook Air  Ultrabook    13.3   
9            10    Acer               Swift 3  Ultrabook    14.0   
...         ...     ...                   ...        ...     ...   
1236       1254      HP         EliteBook 840  Ultrabook    14.0   
1241       1259    Asus  ZenBook UX310UA-WB71  Ultrabook    13.3   
1261       1279    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   
1275       1293    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   
1289       1307    Asus  ZenBook UX305CA-UBM1  Ultrabook    13.3   

                        ScreenResolution                         Cpu  Ram  \
0     IPS Panel Retina Display 2560x1600        Intel Core i5 2.3GHz  8GB   
1                               1440x900        Intel Core i5 1.8GHz  8GB   
4     IPS Panel Retina Display 2560x1600        Intel Core i5 3.1GHz  8GB   
7                               1440x900        Intel Core i5 1.8GHz  8GB   
9            IPS Panel Full HD 1920x1080  Intel Core i5 8250U 1.6GHz  8GB   
...                                  ...                         ...  ...   
1236                           2560x1440  Intel Core i7 6500U 2.5GHz  8GB   
1241                   Full HD 1920x1080  Intel Core i7 6500U 2.5GHz  8GB   
1261         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz  8GB   
1275         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz  8GB   
1289         IPS Panel Full HD 1920x1080    Intel Core M 6Y30 0.9GHz  8GB   

                   Memory                           Gpu       OpSys  Weight  \
0               128GB SSD  Intel Iris Plus Graphics 640       macOS  1.37kg   
1     128GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
4               256GB SSD  Intel Iris Plus Graphics 650       macOS  1.37kg   
7     256GB Flash Storage        Intel HD Graphics 6000       macOS  1.34kg   
9               256GB SSD        Intel UHD Graphics 620  Windows 10   1.6kg   
...                   ...                           ...         ...     ...   
1236            256GB SSD         Intel HD Graphics 520   Windows 7  1.54kg   
1241            256GB SSD         Intel HD Graphics 520  Windows 10  1.45kg   
1261            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   
1275            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   
1289            512GB SSD         Intel HD Graphics 515  Windows 10   1.2kg   

      Price_euros  
0         1339.69  
1          898.94  
4         1803.60  
7         1158.70  
9          770.00  
...           ...  
1236      2198.19  
1241      1280.00  
1261       729.00  
1275       729.00  
1289       729.00  

[140 rows x 13 columns]

In [157]:
df[(df['TypeName'] == 'Ultrabook') & (df['Ram'] == '8GB')].shape

(140, 13)

In [158]:
df[(df['TypeName'] == 'Ultrabook') & (df['Ram'] == '8GB')].shape[0]

140

In [163]:
GB =df[(df['TypeName'] == 'Ultrabook') & (df['Ram'] == '8GB')].shape[0]
GB

140

In [165]:
bezGB = df[df['TypeName'] == 'Ultrabook'].shape[0]
bezGB

196

In [172]:
itog = (GB/bezGB)*100
itog

71.42857142857143

2.5
1. Выберите ноутбук клиенту.
Клиент хочет подобрать ноутбук с 8GB или 16GB ОЗУ на Windows 10 в стоимости до 500 евро, сколько у него вариантов?
2. Выберите ноутбук клиенту.
Клиент хочет подобрать ноутбук от MSI, с видеокартой Nvidia GeForce GTX 1050 Ti и главное не с диагональю 15.6. В какой ценовой категории вышли подобные ноутбуки?
3. Что дешевле?
В среднем дешевле ноутбуки с CPU Intel Core i7 7700HQ 2.8GHz или с Intel Core i7 7600U 2.8GHz?

In [175]:
df[df['OpSys'] == 'Windows 10'].shape

(1072, 13)

In [181]:
df[(df['OpSys'] == 'Windows 10') & (df['Ram'] == '8GB') & (df['Ram'] == '16GB')].shape

(0, 13)

In [205]:
do500Euro = df[(df['OpSys'] == 'Windows 10') & (df['Price_euros'] <= 500.0)]
do500Euro.shape[0]

165

In [206]:
GB8 = df[(df['OpSys'] == 'Windows 10') & (df['Ram'] == '8GB') & (df['Price_euros'] <= 500.0)].shape[0]
GB8

13

In [208]:
GB16 = df[(df['OpSys'] == 'Windows 10') & (df['Ram'] == '16GB') & (df['Price_euros'] <= 500.0)].shape[0]
GB16

0

In [209]:
GB8 + GB16

13